diff --git a/generate.py b/generate.py
index 6f369b3..7334b52 100644
--- a/generate.py
+++ b/generate.py
@@ -705,15 +705,14 @@ def synth(z):
     return clamp_with_grad(model.decode(z_q).add(1).div(2), 0, 1)
 
 
-#@torch.no_grad()
-@torch.inference_mode()
+@torch.no_grad()
 def checkin(i, losses):
     losses_str = ', '.join(f'{loss.item():g}' for loss in losses)
     tqdm.write(f'i: {i}, loss: {sum(losses).item():g}, losses: {losses_str}')
     out = synth(z)
     info = PngImagePlugin.PngInfo()
     info.add_text('comment', f'{args.prompts}')
-    TF.to_pil_image(out[0].cpu()).save(args.output, pnginfo=info) 	
+    TF.to_pil_image(out[0].cpu()).save(args.output % i, pnginfo=info)
 
 
 def ascend_txt():
@@ -749,8 +748,7 @@ def train(i):
     loss.backward()
     opt.step()
     
-    #with torch.no_grad():
-    with torch.inference_mode():
+    with torch.no_grad():
         z.copy_(z.maximum(z_min).minimum(z_max))
 
 
diff --git a/requirements.txt b/requirements.txt
index 1e416b9..e81a630 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -41,19 +41,20 @@ pyasn1-modules==0.2.8
 pyDeprecate==0.3.0
 Pygments==2.9.0
 pyparsing==2.4.7
-pytorch-lightning==1.3.7.post0
+pytorch-lightning
 PyYAML==5.4.1
 regex==2021.4.4
 requests==2.25.1
 requests-oauthlib==1.3.0
 rsa==4.7.2
 six==1.16.0
-tensorboard==2.4.1
-tensorboard-plugin-wit==1.8.0
-torch==1.9.0+cu111
-torchaudio==0.9.0
-torchmetrics==0.3.2
-torchvision==0.10.0+cu111
+# tensorboard==2.4.1
+# tensorboard-plugin-wit==1.8.0
+# torch
+# torchaudio
+# torchmetrics
+# torchvision
+torch-optimizer
 tqdm==4.61.1
 traitlets==5.0.5
 typing-extensions==3.10.0.0
